### PR TITLE
arm/stm32f401rc-rs485: Add support to WS2812 addressable LED

### PR DESCRIPTION
**Pull request by rcsim**
## Summary
This commit adds support to WS2812 on stm32f401rc-rs485 boards.

## Impact
This change impacts only stm32f401rc-rs485 board, adding the possibility of using an addressable LED compatible with WS2812.
It also updates the documentation for  stm32f401rc-rs485 accordingly.

## Testing
Steps used for testing:

1. Configure stm32f401rc-rs485: ws2812.
2. Build and flash the board
3. Access NSH and run ws2812 example application.

It was tested using Ubuntu 22.04.5 LTS on an x86 machine.

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



The PR summary is missing some key information:

* **What functional part of the code is being changed?**  (e.g., specific driver, board configuration, etc.)
* **How does the change exactly work?** (Briefly describe the technical implementation. Did you add a new driver? Modify an existing one?)

**Impact:** 

* The impact section is well-detailed.

**Testing:**

* **Insufficient information:**  
    * You need to provide the actual testing logs before and after the change.  "your testing logs here" is not helpful.
    * Be specific about the compiler version used.
    * "ws2812 example application" -  Provide the full path to this application within the NuttX apps repository. 

**To summarize:** The PR needs more detail in the summary and testing sections to meet the NuttX requirements fully.

**Review by acassis — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/stm32/stm32f401rc-rs485/src/stm32_bringup.c [hunk 1]
```diff
@@ -328,5 +332,15 @@ int stm32_bringup(void)
     }
 #endif
 
+#ifdef CONFIG_WS2812 && defined(CONFIG_WS2812_LED_COUNT)
```
Comment (pkarashchenko) [inline boards/arm/stm32/stm32f401rc-rs485/src/stm32_bringup.c:335]:
```suggestion
#if defined(CONFIG_WS2812) && defined(CONFIG_WS2812_LED_COUNT)
```

Comment (rcsim) [reply boards/arm/stm32/stm32f401rc-rs485/src/stm32_bringup.c]: Hi @pkarashchenko,
Thanks for your suggestion, I've pushed a new patch with your suggestion included.

**Comment by acassis** [comment]
@rcsim please rebase your branch and git push forced to fix the CI issue

**Comment by acassis** [comment]
@xiaoxiang781216 @lupyuen what are these errors:

```
Cmake in present: nucleo-f722ze/cansock,CONFIG_ARM_TOOLCHAIN_CLANG
Configuration/Tool: nucleo-f722ze/cansock,CONFIG_ARM_TOOLCHAIN_CLANG
2024-10-21 23:03:46
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Select HOST_LINUX=y
CMake Error at cmake/nuttx_add_library.cmake:136 (add_library):
  Cannot find source file:

    /github/workspace/sources/nuttx/boards/arm/stm32f7/common/src/stm32_cansock.c
Call Stack (most recent call first):
  boards/CMakeLists.txt:34 (nuttx_add_kernel_library)


CMake Generate step failed.  Build files cannot be regenerated correctly.
-- Configuring done (0.3s)
CMake Error at cmake/nuttx_add_library.cmake:136 (add_library):
  Cannot find source file:

    /github/workspace/sources/nuttx/boards/arm/stm32f7/common/src/stm32_cansock.c
Call Stack (most recent call first):
  boards/CMakeLists.txt:34 (nuttx_add_kernel_library)


-- Generating done (0.2s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_CLANG
  Building NuttX...
ninja: error: loading 'build.ninja': No such file or directory
ninja: error: loading 'build.ninja': No such file or directory
cp: cannot stat 'nuttx.bin': No such file or directory
cp: cannot stat 'nuttx.bin': No such file or directory
ninja: error: loading 'build.ninja': No such file or directory
ninja: error: loading 'build.ninja': No such file or directory
```

**Review by linguini1 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@xuxin930 could you look at why cmake output the above warning.

**Pull request merged**